### PR TITLE
Add (interactive and non-interactive) simulator and support of `DiscreteProblem`; Add FSimROS

### LINKED ISSUE #142: ROS2 example
Add an example of ROS2-compatible scripts

### LINKED ISSUE #143: Add (interactive and non-interactive) simulator
See [FSimBase.jl, #5](https://github.com/JinraeKim/FSimBase.jl/issues/5).

### PR DESCRIPTION
**Pull request by JinraeKim**
See [FSimBase.jl, #6](https://github.com/JinraeKim/FSimBase.jl/pull/6).

Note that [FSimROS.jl](https://github.com/JinraeKim/FSimROS.jl) is a new related package for ROS2.

### CODE REVIEW

**Comment by hnlee77** [comment]
이거 figures도 함께 올린거 맞나요?

**Comment by JinraeKim** [comment]
> 이거 figures도 함께 올린거 맞나요?

네 (README 때문)

File: README.md [hunk 1]
```diff
@@ -97,11 +96,16 @@ function test()
...
+    display(fig)
     df
 end
+
+@testset "lqr example" begin
+    main()
+end
 ```
 
 ```julia
```
Comment (hnlee77) [inline README.md:161]: 이 부분은 main()로 변경이 필요하겠네요 ㅎㅎ 급하진 않지만

**Review by hnlee77 — APPROVED** [review]
예시 잘 봤습니다. 감사합니다. ㅎㅎ

File: README.md [hunk 2]
```diff
@@ -123,17 +123,94 @@ julia> test()
...
-![ex_screenshot](./figures/x_lqr.png)
-![ex_screenshot](./figures/u_lqr.png)
+![ex_screenshot](./figures/lqr.png)
+
+### Linear system with zero-order-hold (ZOH) input
+- Note that this example utilises interactive simulation interface. See `test/environments/integrated_environments/linear_system_zoh_input.jl`.
+
+```julia
+using FlightSims
+import FlightSims
```
Comment (hnlee77) [inline README.md]: 혹시 여기서 using 을 하고 import를 하는건 어떤 의미일까요?

**Comment by JinraeKim** [comment]
Update compat for [the bug fix of `SimulationLogger`](https://github.com/JinraeKim/SimulationLogger.jl/pull/8) if necessary.

**Comment by JinraeKim** [comment]
~~@hnlee77 @nhcho91~~
~~To control saving and step behaviours in detail, I detached `df` argument from `step!` and `step_until!`.~~
~~Instead, use `push!`.~~
~~See the revised `README.md`.~~

~~### Why is it necessary?~~
~~As shown in `test/minimal.jl`, I'd like to reject "a state which is out of a given bound"~~
Wrong place; I'm sorry. Related to [FSimBase.jl, #6, comment](https://github.com/JinraeKim/FSimBase.jl/pull/6#issuecomment-940752299).

**Comment by JinraeKim** [comment]
@hnlee77 @nhcho91 
Now it supports `DiscreteProblem`.
Please refer to the readme and related PRs, e.g., [FSimBase, #6](https://github.com/JinraeKim/FSimBase.jl/pull/6).

Please feel free to report any related feature requests and issues :)

**Pull request merged**